src/main/java/org/carlspring/cloud/storage/s3fs/S3FileSystemProvider.java

```diff
@@ -502,23 +573,32 @@ public void createDirectory(Path dir,
         }
 
         // create bucket if necessary
-        Bucket bucket = s3Path.getFileStore().getBucket();
-        String bucketName = s3Path.getFileStore().name();
+        final Bucket bucket = s3Path.getFileStore().getBucket();
+        final String bucketName = s3Path.getFileStore().name();
+
         if (bucket == null)
         {
-            s3Path.getFileSystem().getClient().createBucket(bucketName);
+            //TODO: How to get s3fs_region property value?
+            final Region region = Region.US_EAST_1;
+            final CreateBucketConfiguration configuration = CreateBucketConfiguration.builder()
+                                                                                     .locationConstraint(region.id())
+                                                                                     .build();
+            final CreateBucketRequest request = CreateBucketRequest.builder()
+                                                                   .bucket(bucketName)
+                                                                   .createBucketConfiguration(configuration)
```

Comment:
Unless it is compulsory to have a CreateBucketConfiguration I would leave this out as it is only setting the region (see above)